.circleci/config.yml

```diff
@@ -32,11 +32,26 @@ jobs:
         chmod +x ./architect
         ./architect version
 
+    - run: |
+        date +"%Y" > /tmp/year
+        date +"%m" > /tmp/month
+        date +"%d" > /tmp/day
+    - restore_cache:
+        keys:
```

Comment:
Cache keys are prefix matched in order, so we attempt to use a go cache from today, this month, and then this year in order.
This does mean the first build of a day / month / year will be somewhat slower, but I expect it will still be faster overall.